wal/wal.go

```diff
@@ -532,21 +540,28 @@ func (w *WAL) saveState(s *raftpb.HardState) error {
 }
 
 func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
-	w.mu.Lock()
```

Comment:
revert the changes in this func? let's first improve save entry by saving one allocation? then we can move the encoding out of lock?